back/app/models/idea.rb

```diff
@@ -220,6 +222,34 @@ def fix_comments_count_on_projects
   def update_phase_ideas_count(_)
     IdeasPhase.counter_culture_fix_counts only: %i[phase]
   end
+
+  def validate_creation_phase
+    return unless creation_phase
+
+    if creation_phase.project_id != project_id
+      errors.add(
+        :creation_phase,
+        :invalid_project,
+        message: 'The creation phase must be a phase of the input\'s project'
+      )
+    end
+
+    if %w[ideation budgeting].include?(creation_phase.participation_method)
```

Comment:
When we write conditionals based on the participation method, we should delegate to the participation method object. Something like:
```
participation_method_on_creation.transitive?
```